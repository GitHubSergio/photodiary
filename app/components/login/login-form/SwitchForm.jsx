// Imports
import React from 'react';
import { Button } from 'react-native';
import { useNavigation } from '@react-navigation/native';

const SwitchForm = ({ route }) => {
  const navigation = useNavigation();
  const goTo = route === 'LoginScreen' ? 'SignUpScreen' : 'LoginScreen';
  const label = route === 'LoginScreen' ? 'Sign Up' : 'Log In';

  return <Button testID="switch-btn" onPress={() => navigation.navigate(goTo)} title={label} />;
};

export default SwitchForm;
