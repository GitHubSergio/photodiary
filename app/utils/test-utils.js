/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { Provider } from 'react-redux';
import { applyMiddleware, createStore } from 'redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import rootReducer from '../redux/reducers';
import { render } from '@testing-library/react-native';
// import { render } from 'react-native-testing-library';
import reduxStoreStateMocked from './redux-state';

export const renderWithReduxMockedStore = (ui, updatedStore) => {
  const middlewares = [thunk];
  const mockStore = configureMockStore(middlewares);
  const store = mockStore({ ...reduxStoreStateMocked, ...updatedStore });
  // console.log(store.getState());
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
  };
};

export const renderWithRedux = (
  ui,
  {
    initialState,
    store = createStore(rootReducer, initialState, applyMiddleware(thunk)),
  } = {},
) => ({
  ...render(<Provider store={store}>{ui}</Provider>),
  store,
});
