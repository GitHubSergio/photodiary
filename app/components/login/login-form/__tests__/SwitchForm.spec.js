/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { renderWithRedux } from 'test-utils';
import { useRoute, useNavigation } from '@react-navigation/native';
import { render, fireEvent } from 'react-native-testing-library';
import SwitchForm from '../SwitchForm';

jest.mock('@react-navigation/native', () => ({
  useRoute: jest.fn(() => ({
    name: 'SignInScreen',
  })),
  useNavigation: jest.fn(),
}));

/*describe('<SwitchForm />', () => {
  it('should return the label Sign Up if in the LoginScreen', () => {
    const { toJSON } = render(<SwitchForm route="LoginScreen" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should return the label Log In if in the SignUpScreen', () => {
    const { toJSON } = render(<SwitchForm route="SignUpScreen" />);
    expect(toJSON()).toMatchSnapshot();
  });
});*/
