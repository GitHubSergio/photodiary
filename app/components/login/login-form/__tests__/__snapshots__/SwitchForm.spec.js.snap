// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SwitchForm /> should return the label Log In if in the SignUpScreen 1`] = `
<Button
  onPress={[Function]}
  title="Log In"
/>
`;

exports[`<SwitchForm /> should return the label Sign Up if in the LoginScreen 1`] = `
<Button
  onPress={[Function]}
  title="Sign Up"
/>
`;
